### PR TITLE
12-shoppingcart_and_cartitem_test_cases (done)

### PR DESCRIPTION
**Pull request by andrey-kiyas**
Added:
- ShoppingCart and CartItem test cases
  (repository, service, controller)

### CODE REVIEW

**Comment by andrey-kiyas** [comment]
image

File: src/main/java/com/onlinebookstore/config/MapperConfig.java [hunk 1]
```diff
@@ -10,4 +10,5 @@
         implementationPackage = "<PACKAGE_NAME>.impl"
 )
 public class MapperConfig {
+
```
Comment (Rommelua) [inline src/main/java/com/onlinebookstore/config/MapperConfig.java]: ??

File: src/test/java/com/onlinebookstore/controller/ShoppingCartControllerTest.java [hunk 2]
```diff
@@ -0,0 +1,224 @@
...
+            );
+        }
+    }
+
+    @Order(1)
+    @Test
+    @DisplayName("Get shopping cart")
+    void getShoppingCart_ValidRequest_Success() throws Exception {
+        // Given
+        User mockUser = getMockUser();
```
Comment (Rommelua) [inline src/test/java/com/onlinebookstore/controller/ShoppingCartControllerTest.java:89]: you can use annotation
image

File: src/test/java/com/onlinebookstore/service/impl/ShoppingCartServiceImplTest.java [hunk 3]
```diff
@@ -0,0 +1,334 @@
...
+        book3.setCategories(new HashSet<>());
+        book3.setCoverImage("Cover Image");
+        book3.setDeleted(true);
+        book3.setDescription("The characteristics of someone or something");
+        book3.setId(1L);
+        book3.setIsbn("Isbn");
+        book3.setPrice(BigDecimal.valueOf(1L));
+        book3.setTitle("Dr");
+
+        User user4 = new User();
```
Comment (Rommelua) [inline src/test/java/com/onlinebookstore/service/impl/ShoppingCartServiceImplTest.java]: do we need 4 users to test updateCartItem method?

File: src/test/java/com/onlinebookstore/service/impl/ShoppingCartServiceImplTest.java [hunk 4]
```diff
@@ -0,0 +1,334 @@
...
+    private UserRepository userRepository;
+
+    @Test
+    public void testGetShoppingCartDtoByUserId() {
+        User user = new User();
+        user.setDeleted(true);
+        user.setEmail("jane.doe@example.org");
+        user.setFirstName("Jane");
+        user.setId(1L);
+        user.setLastName("Doe");
```
Comment (Rommelua) [inline src/test/java/com/onlinebookstore/service/impl/ShoppingCartServiceImplTest.java:61]: You do many extra work. See how we can test Book for example
image

**Review by Rommelua — CHANGES_REQUESTED** [review]

File: src/test/java/com/onlinebookstore/service/impl/ShoppingCartServiceImplTest.java (hunk 3 above)
Comment (andrey-kiyas) [reply src/test/java/com/onlinebookstore/service/impl/ShoppingCartServiceImplTest.java]: No, they are not needed. It was an unnecessary flight of fancy. Corrected.

File: src/main/java/com/onlinebookstore/config/MapperConfig.java (hunk 1 above)
Comment (andrey-kiyas) [reply src/main/java/com/onlinebookstore/config/MapperConfig.java]: Done

File: src/test/java/com/onlinebookstore/controller/ShoppingCartControllerTest.java (hunk 2 above)
Comment (andrey-kiyas) [reply src/test/java/com/onlinebookstore/controller/ShoppingCartControllerTest.java:89]: Yes, this is also possible, but I thought my version was more readable and simpler.

File: src/test/java/com/onlinebookstore/service/impl/ShoppingCartServiceImplTest.java (hunk 4 above)
Comment (andrey-kiyas) [reply src/test/java/com/onlinebookstore/service/impl/ShoppingCartServiceImplTest.java:61]: I did "ShoppingCartServiceImplTest" according to the chosen method in the first part of the test module; I covered only "ShoppingCartControllerTest" with integration testing.

**Pull request merged**